frontend/src/pages/profile/Scheduler.tsx

```diff
@@ -0,0 +1,148 @@
+import { Accordion, AccordionHeader, AccordionBody } from '@material-tailwind/react';
+import { useEffect, useState } from 'react';
+import { ChevronDownIcon, ChevronUpIcon } from '@heroicons/react/24/solid';
+import SchedulerHeader from './SchedulerHeader';
+import SchedulerRow from './SchedulerRow';
+import SchedulerControl from './SchedulerControl';
+import type { Availability, SchedulerRowItem } from '../dashboard';
+import dayjs from 'dayjs';
+import { AvailabilityType } from '@/common';
+
+const Scheduler = ({
+  name,
+  availability,
+  onChangeAvailabilityDates,
+}: {
+  name: string;
+  availability?: Availability[];
+  onChangeAvailabilityDates: (from: string, to: string) => void;
+}) => {
+  const [open, setOpen] = useState(1);
+  const handleOpen = (value: number) => setOpen(open === value ? 0 : value);
+  const [schedulerRows, setSchedulerRows] = useState<{
+    [key: string]: SchedulerRowItem[];
+  }>();
+
+  useEffect(() => {
+    if (availability) {
```

Comment:
Can we set a default value for availability to avoid checking if exists?

Reply:
Oh right, I forgot to take this out. We are defaulting to [] now from `usePersonnel`